src/actuator/ActuatorParsingFAST.C

```diff
@@ -0,0 +1,186 @@
+
+  get_required(turbNode, "turbine_base_pos", fi.globTurbineData[iTurb].TurbineBasePos);
+  get_required(turbNode, "turbine_hub_pos" , fi.globTurbineData[iTurb].TurbineHubPos );
+  get_required(
+    turbNode, "num_force_pts_blade",
+    fi.globTurbineData[iTurb].numForcePtsBlade);
+  get_required(
+    turbNode, "num_force_pts_tower", fi.globTurbineData[iTurb].numForcePtsTwr);
+  fi.globTurbineData[iTurb].nacelle_cd = 0.0;
+  fi.globTurbineData[iTurb].nacelle_area = 0.0;
+  fi.globTurbineData[iTurb].air_density = 0.0;
+  get_if_present(turbNode, "nacelle_cd", fi.globTurbineData[iTurb].nacelle_cd);
+  get_if_present(
+    turbNode, "nacelle_area", fi.globTurbineData[iTurb].nacelle_area);
+  get_if_present(
+    turbNode, "air_density", fi.globTurbineData[iTurb].air_density);
+  //TODO(psakiev) add unit test to make sure this doesn't change with the openfast API
+  actMetaFAST.numPointsTurbine_.h_view(iTurb) = 1 // hub
+      + fi.globTurbineData[iTurb].numForcePtsTwr
+      + fi.globTurbineData[iTurb].numForcePtsBlade * 3; //TODO(psakiev) double check blade num specification
+}
+}
+
+ActuatorMetaFAST actuator_FAST_parse(const YAML::Node& y_node, const ActuatorMeta& actMeta){
+  ActuatorMetaFAST actMetaFAST(actMeta);
+  fast::fastInputs& fi = actMetaFAST.fastInputs_;
+  fi.comm = NaluEnv::self().parallel_comm();
+  fi.nTurbinesGlob = actMetaFAST.numberOfActuators_;
+
+  const YAML::Node y_actuator = y_node["actuator"];
+  ThrowErrorMsgIf(!y_actuator,"actuator argument is "
+    "missing from yaml node passed to actuator_FAST_parse");
+  if(fi.nTurbinesGlob > 0){
+    get_if_present(y_actuator, "dry_run", fi.dryRun, false);
+    get_if_present(y_actuator, "debug", fi.debug, false);
+    get_required(y_actuator, "t_start", fi.tStart);
+    std::string simStartType = "na";
+    get_required(y_actuator, "simStart", simStartType);
+    if (simStartType == "init") {
+      if (fi.tStart == 0) {
+        fi.simStart = fast::init;
+      } else {
+        throw std::runtime_error(
+          "actuators: simStart type not consistent with start time for FAST");
+      }
+    } else if (simStartType == "trueRestart") {
+      fi.simStart = fast::trueRestart;
+    } else if (simStartType == "restartDriverInitFAST") {
+      fi.simStart = fast::restartDriverInitFAST;
+    }
+    get_required(y_actuator, "n_every_checkpoint", fi.nEveryCheckPoint);
+    get_required(y_actuator, "dt_fast", fi.dtFAST);
+    get_required(y_actuator, "t_max", fi.tMax);
+
+    if (y_actuator["super_controller"]) {
+      get_required(y_actuator, "super_controller", fi.scStatus);
+      get_required(y_actuator, "sc_libFile", fi.scLibFile);
+      get_required(y_actuator, "num_sc_inputs", fi.numScInputs);
+      get_required(y_actuator, "num_sc_outputs", fi.numScOutputs);
+    }
+
+    fi.globTurbineData.resize(fi.nTurbinesGlob);
+
+    for (int iTurb = 0; iTurb < fi.nTurbinesGlob; iTurb++) {
+      if (y_actuator["Turbine" + std::to_string(iTurb)]) {
```

Comment:
Why hardcode turbine name pattern?

Reply:
This is just copied and pasted from the current code.  It is how the yaml nodes are specified. The name is actually another parameter inside the node, but I can look at how to change this and make them the same.